app/controllers/api/providers_controller.rb

```diff
@@ -131,6 +114,36 @@ def change_password_resource(type, id, data = {})
 
     private
 
+    def provider_options(type)
+      klass = type.safe_constantize
+
+      raise BadRequestError, "Invalid provider - #{type}" unless klass.try(:<, ExtManagementSystem)
+      raise BadRequestError, "No DDF specified for - #{type}" unless klass.respond_to?(:params_for_create)
+
+      klass.params_for_create
+    end
+
+    def providers_options
+      providers_options = ManageIQ::Providers::BaseManager.leaf_subclasses.inject({}) do |po, ems|
+        po.merge(ems.ems_type => ems.options_description)
+      end
+
+      supported_providers = ExtManagementSystem.supported_types_for_create.map do |klass|
+        if klass.supports_regions?
+          regions = klass.parent::Regions.all.sort_by { |r| r[:description] }.map { |r| r.slice(:name, :description) }
+        end
+
+        {
+          :title   => klass.description,
+          :type    => klass.to_s,
+          :kind    => klass.to_s.demodulize.sub(/Manager$/, '').underscore,
+          :regions => regions
+        }.compact
+      end
+
+      { "provider_settings" => providers_options, "supported_providers" => supported_providers }
```

Comment:
If someone asks for a specific provider type could we return all of this in both cases for consistency rather than this or [just the params_for_create](https://github.com/ManageIQ/manageiq-api/pull/645/files#diff-14a4ff3ca25095b0ebbc061c5aebf20dR123) depending on if they passed a specific provider type or not?

Reply:
If you ask for a single type you should get a subset of what you would get if you asked for all of them

Reply:
I'll see what I can do :wink:

Reply:
Updated!